source/auth/tests/connection-string.yml

```diff
@@ -2,366 +2,336 @@ tests:
     -
         description: "should use the default source and mechanism"
         uri: "mongodb://user:password@localhost"
-        hosts: ~
         valid: true
-        warning: false
-        auth:
+        credential_configured: true
+        credential:
             username: "user"
             password: "password"
-            db: "admin"
-        options: ~
+            source: "admin"
+            mechanism: "default"
+            mechanism_properties: {}
     -
         description: "should use the database when no authSource is specified"
         uri: "mongodb://user:password@localhost/foo"
-        hosts: ~
         valid: true
-        warning: false
-        auth:
+        credential_configured: true
+        credential:
             username: "user"
             password: "password"
-            db: "foo"
-        options: ~
+            source: "foo"
+            mechanism: "default"
+            mechanism_properties: {}
     -
         description: "should use the authSource when specified"
         uri: "mongodb://user:password@localhost/foo?authSource=bar"
-        hosts: ~
         valid: true
-        warning: false
-        auth:
+        credential_configured: true
+        credential:
             username: "user"
             password: "password"
-            db: "bar"
-        options: ~
+            source: "bar"
+            mechanism: "default"
+            mechanism_properties: {}
     -
         description: "should recognise the mechanism (GSSAPI)"
         uri: "mongodb://user%40DOMAIN.COM@localhost/?authMechanism=GSSAPI"
-        hosts: ~
         valid: true
-        warning: false
-        auth:
+        credential_configured: true
+        credential:
             username: "user@DOMAIN.COM"
-            password: ~
-            db: "$external"
-        options:
-            authmechanism: "GSSAPI"
+            password: ""
+            source: "$external"
+            mechanism: "GSSAPI"
+            mechanism_properties:
+                SERVICE_NAME: "mongodb"
     -
         description: "should ignore the database (GSSAPI)"
         uri: "mongodb://user%40DOMAIN.COM@localhost/foo?authMechanism=GSSAPI"
-        hosts: ~
         valid: true
-        warning: false
-        auth:
+        credential_configured: true
+        credential:
             username: "user@DOMAIN.COM"
-            password: ~
-            db: "$external"
-        options:
-            authmechanism: "GSSAPI"
+            password: ""
+            source: "$external"
+            mechanism: "GSSAPI"
+            mechanism_properties:
+                SERVICE_NAME: "mongodb"
```

Comment:
Same

Reply:
I think this can be resolved in light of https://github.com/mongodb/specifications/pull/532/files#r278946134